### ISSUE TITLE
Unique together constraint/rule

### ISSUE THREAD

**Issue by blackmann** [pocketbase/pocketbase#544]
I'm struggling to figure out how to add a UNIQUE TOGETHER constraint/rule for `create`/`update` action.

Elaboration:

Say I have an ownership table with fields: `user`, `page`, `role` and I want `user` and `page` values to be unique together, how do I specify the rule?

Or perhaps the collection form can accept a unique together field which will accept a list of UNIQUE TOGETHER fields - which will be used to create the SQL constraint query.

**Comment by ganigeorgiev** [comment]
This was recently discussed in https://github.com/pocketbase/pocketbase/discussions/543#discussioncomment-3677076.

At the moment, from the interface you can define only a unique constrain for a single field.

You have to either use the OnModelBefore* [event hooks](https://pocketbase.io/docs/event-hooks/#db-hooks) or define manually (or via migration) a sql unique index across multiple columns:

```go
app.Dao().DB().NewQuery(`
    CREATE UNIQUE INDEX ownership_user_page_idx ON ownership (user, page);
`).Execute();
```

Since this was requested previously several times, I'll consider it and add it to the roadmap (maybe combined with https://github.com/pocketbase/pocketbase/issues/345).

**Comment by benallfree** [comment]
Tagging #542 in here just in case it makes sense to cover it at the same time.

**Comment by tt9** [comment]
Is it possible to simulate this with filter rules?

**Comment by ganigeorgiev** [comment]
@tt9 I don't think so but it depends on your use case and the type of fields you want to restrict. The easiest workaround for now is to execute manually the "create index" statement directly against the database.

**Comment by ganigeorgiev** [comment]
Supports for unique constraints across multiple fields (including expression collumns and partial indexes, aka. conditions) was implemented in the `develop` branch and will be available with the next v0.14.0 release (most likely at the end of the month or the first week of April).

The UI currently looks like this:

![main](https://i.imgur.com/Tf0QMA6.png)

![popup](https://i.imgur.com/tcpG2WO.png)
